recipe/migrations/numpy2.yaml

```
@@ -0,0 +1,75 @@
+__migrator:
+  build_number: 1
+  kind: version
+  commit_message: |
+    Rebuild for numpy 2.0
+    
+    TL;DR: The way we build against numpy has changed as of numpy 2.0. This bot
+    PR has updated the recipe to account for the changes (see below for details).
+    The numpy 2.0 package itself is currently only available from a special release
+    channel (`conda-forge/label/numpy_rc`) and will not be available on the main
+    `conda-forge` channel until the release of numpy 2.0 GA.
+    
+    The biggest change is that we no longer need to use the oldest available numpy
+    version at build time in order to support old numpy version at runtime - numpy
+    will by default use a compatible ABI for the oldest still-supported numpy versions.
+    
+    Additionally, we no longer need to use `{{ pin_compatible("numpy") }}` as a
+    run requirement - this has been handled for more than two years now by a
+    run-export on the numpy package itself. The migrator will therefore remove
+    any occurrences of this.
```

Maybe we can condense these to provide a clear takeaway?

```suggestion
    The biggest change is that we no longer need to use the oldest available NumPy
    version at build time in order to support old NumPy versions at runtime. Instead
    NumPy will default to using the oldest compatible NumPy ABI available. So we
    drop `{{ pin_compatible("numpy") }}` from `requirements/run`. This also is
    not needed for `numpy` version `1.19`+ as it is in `numpy`'s own `run_exports`.
```

---

The "So we drop" implies that dropping of `pin_compatible` being due to numpy's new default ABI compatibility, but this is not the case. If `pin_compatible` supported overriding the `lower_bound=` and numpy had no run-export, there would be no reason to remove it. So I'd like to keep these things separate.

---

Tried to put together a suggestion instead of just saying let's shorten this, but am open to other approaches one might take if you have ideas

---

I want to keep it as-is, unless there's a better (and equally accurate) suggestion.